src/framework/components/anim/controller.js

```diff
@@ -0,0 +1,560 @@
+                sum += this.animations[i].weight;
+            }
+            return sum;
+        },
+
+        getTimelineDuration: function () {
+            var duration = 0;
+            var i;
+            for (i = 0; i < this.animations.length; i++) {
+                var animation = this.animations[i];
+                if (animation.animTrack.duration > duration) {
+                    duration = animation.animTrack.duration > duration;
+                }
+            }
+            return duration;
+        }
+    });
+
+    var AnimTransition = function (controller, from, to, time, priority, conditions, exitTime, transitionOffset, interruptionSource) {
+        this.controller = controller;
+        this.from = from;
+        this.to = to;
+        this.time = time;
+        this.priority = priority;
+        this.conditions = conditions || [];
+        this.exitTime = exitTime || null;
+        this.transitionOffset = transitionOffset || null;
+        this.interruptionSource = interruptionSource || ANIM_INTERRUPTION_SOURCE_NONE;
+    };
+
+    Object.assign(AnimTransition.prototype, {
+        hasConditionsMet: function () {
+            var conditionsMet = true;
+            var i;
+            for (i = 0; i < this.conditions.length; i++) {
+                var condition = this.conditions[i];
+                var parameter = this.controller._getParameter(condition.parameterName);
+                switch (condition.predicate) {
+                    case ANIM_TRANSITION_PREDICATE_GREATER_THAN:
+                        conditionsMet = conditionsMet && parameter.value > condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_LESS_THAN:
+                        conditionsMet = conditionsMet && parameter.value < condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_GREATER_THAN_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value >= condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_LESS_THAN_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value <= condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value === condition.value;
+                        break;
+                    case ANIM_TRANSITION_PREDICATE_NOT_EQUAL_TO:
+                        conditionsMet = conditionsMet && parameter.value !== condition.value;
+                        break;
+                }
+                if (!conditionsMet)
+                    return conditionsMet;
+            }
+            return conditionsMet;
+        },
+        hasExitTime: function () {
+            return !!this.exitTime;
+        }
+    });
+
+    var AnimController = function (animEvaluator, states, transitions, parameters, activate) {
```

Comment:
If AnimController is going to be public API we should prefix private variables with `_`

Reply:
Done